WordPress/Classes/Models/Blog+Lookup.swift

```diff
@@ -58,4 +69,79 @@ public extension Blog {
 
         return service.findBlog(withXmlrpc: xmlrpc, andUsername: username)
     }
+
+    @objc(countInContext:)
+    static func count(in context: NSManagedObjectContext) -> Int {
+        BlogQuery().count(in: context)
+    }
+
+    @objc(wpComBlogCountInContext:)
+    static func wpComBlogCount(in context: NSManagedObjectContext) -> Int {
+        BlogQuery().hostedByWPCom(true).count(in: context)
+    }
+
+    @objc(selfHostedInContext:)
+    static func selfHosted(in context: NSManagedObjectContext) -> [Blog] {
+        (try? BlogQuery().hostedByWPCom(false).blogs(in: context)) ?? []
+    }
+}
+
+struct BlogQuery {
+    private var predicates = [NSPredicate]()
+
+    func blogID(_ id: Int) -> Self {
+        blogID(Int64(id))
+    }
+
+    func blogID(_ id: NSNumber) -> Self {
+        blogID(id.int64Value)
+    }
+
+    func blogID(_ id: Int64) -> Self {
+        and(NSPredicate(format: "blogID = %ld", id))
+    }
+
+    func username(_ username: String) -> Self {
+        and(NSPredicate(format: "account.username = %@", username))
+    }
```

Comment:
For reference, here is [the original code](https://github.com/wordpress-mobile/WordPress-iOS/pull/19374/files#diff-5fbf56427ffcef5aa28ecfb7ec87141fb9ff440ef0378f99d5cec0984a4d4fb1L34-L38):
```objc

- (Blog *)blogByBlogId:(NSNumber *)blogID andUsername:(NSString *)username
{
    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"blogID = %@ AND account.username = %@", blogID, username];
    return [self blogWithPredicate:predicate];
}
```